File: java/queues/LinkedDeque.java
package queues;

import fds.DoublyLinkedList;

public class LinkedDeque<E> implements Deque<E> {
    private DoublyLinkedList<E> list = new DoublyLinkedList<>();

    public LinkedDeque() { // an empty list

    }

    @Override
    public void addFirst(E e) {
        list.addFirst(e);
    }

    @Override
    public void addLast(E e) {
        list.addLast(e);
        ;
    }

    @Override
    public E first() {
        return list.first();
    }

    @Override
    public boolean isEmpty() {
        return list.isEmpty();
    }

    @Override
    public E last() {
        return list.last();
    }

    @Override
    public E removeFirst() {
        return list.removeFirst();
    }

    @Override
    public E removeLast() {
        return list.removeLast();
    }

    @Override
    public int size() {
        return list.size();
    }

    @Override
    public String toString() {
        return "LinkedDeque [list=" + list + "]";
    }

}

File: python/A First Course on Data Structures in Python/ds/deque/linkeddeque.py
from ds.linkedlist import LinkedList


class LinkedDeque:
    """ A Deque implemnted using a LinkedList. """

    def __init__(self):
        self._L = LinkedList()

    def add_first(self, item):
        self._L.add_first(item)

    def add_last(self, item):
        self._L.add_last(item)

    def remove_first(self):
        return self._L.remove_first()

    def remove_last(self):
        return self._L.remove_last()
